### PR TITLE
Enable roundeven expansion pattern

### PR DESCRIPTION
**Pull request by jpienaar**
Re-enables more tests from #12778.

### CODE REVIEW

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Could we keep the approximate math patterns still guarded by a flag. Even if the flag is on by default.

**Comment by jpienaar** [comment]
> Could we keep the approximate math patterns still guarded by a flag. Even if the flag is on by default.

Good idea, done.

**Review by rsuderman — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp [hunk 1]
```diff
@@ -35,6 +35,7 @@ class PolynomialApproximationPass
       mathPatterns.add<math::ErfPolynomialApproximation>(&getContext());
     } else {
       populateMathPolynomialApproximationPatterns(mathPatterns);
+      populateExpandRoundEvenPattern(mathPatterns);
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp:38]: These arent exact expansions? Thats surprising.

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp (hunk 1 above)
Comment (jpienaar) [reply compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp:38]: Perhaps we need a different pass here as this one is just a plain expansion but you had asked to guard it, but perhaps we spoke past each other.

**Pull request merged**